microbench/src/main/java/io/netty/microbench/channel/AutoFlushBenchmark.java

```diff
@@ -0,0 +1,242 @@
+import io.netty.channel.ChannelOption;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoop;
+import io.netty.channel.SelectStrategy;
+import io.netty.channel.SelectStrategyFactory;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.microbench.util.AbstractSharedExecutorMicrobenchmark;
+import io.netty.util.IntSupplier;
+import io.netty.util.concurrent.DefaultThreadFactory;
+import io.netty.util.concurrent.Future;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+
+import java.net.SocketAddress;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+
+import static java.util.concurrent.TimeUnit.*;
+
+@State(Scope.Benchmark)
+public class AutoFlushBenchmark extends AbstractSharedExecutorMicrobenchmark {
+
+    @Param({ "true", "false" })
+    public boolean flush;
+
+    @Param({ "1", "10", "100" })
+    public int writeCount;
+
+    private ChannelPipeline pipeline;
+    private ByteBuf payload;
+    private Channel serverChannel;
+    private Channel clientChannel;
+    private NioEventLoopGroup serverEventloop;
+    private NioEventLoopGroup clientEventLoop;
+
+    @Setup(Level.Trial)
+    public void setup() {
+        serverEventloop = new NioEventLoopGroup(1, new DefaultThreadFactory("server", true));
+        clientEventLoop = new NioEventLoopGroup(1, new DefaultThreadFactory("client", true)) {
+            @Override
+            protected EventLoop newChild(Executor executor, Object... args) throws Exception {
+                return super.newChild(executor, args[0], new SelectStrategyFactory() {
+                    @Override
+                    public SelectStrategy newSelectStrategy() {
+                        return new SelectStrategy() {
+
+                            @Override
+                            public int calculateStrategy(IntSupplier selectSupplier, boolean hasTasks)
+                                    throws Exception {
+                                if (hasTasks) {
+                                    return selectSupplier.get();
+                                }
+                                return SELECT;
+                            }
+                        };
+                    }
+                });
+            }
+        };
+        ServerBootstrap sb = new ServerBootstrap();
+        sb.group(serverEventloop)
+          .channel(NioServerSocketChannel.class)
+          .childHandler(new ChannelInitializer<Channel>() {
+              @Override
+              protected void initChannel(Channel ch) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1186)